### DISCUSSION TITLE
Using emoticons in chatbox gives an error (front)

### DISCUSSION THREAD

**Discussion by Alex-e107nl** [e107inc/e107#5127] [Q&A Core] [answered]
Hi, when i activate the emoticons (does not matter which pack, also with default) it gives  https://e107.nl/e107_images/emotes/SkypeSmileys/e-token status 404 on my frontpage with the chatbox plugin (e107.nl). When you are logged in and log out you get directed to /e107_images/emotes/SkypeSmileys/e-token which gives the error page... In the admin it works perfectly (thx @Jimmi08 again [https://github.com/e107inc/e107/issues/4972](https://github.com/e107inc/e107/issues/4972)) Using e107 2.3.3 PHP 8.1

**Comment by Alex-e107nl** [comment]
This problem is still there, it is really the combination of the chat box with enabled emoticons in the chat box, as soon as I turn it off in the admin of the chat box, the message disappears. Github today.

**Reply by Jimmi08** [reply]
Hi, I tried it on e107sk.com, no problem. Default emoticons.

**Comment by Jimmi08** [comment]
Maybe related:
https://github.com/e107inc/e107/issues/5142

**Reply by Jimmi08** [reply]
your url is `https://e107.nl/index.php`  but chatbox post URL is `https://e107.nl/index.php? `

**Reply by Jimmi08** [reply]
my url is `https://www.e107sk.com/news/`  and chatbox post url is `https://www.e107sk.com/news/? `

**Comment by Alex-e107nl** [comment]
On e107.nl removed log plugin, problem with the token search is still there, also removed eu cookie plugin...

**Comment by Alex-e107nl** [comment]
Still no solution, there is also an e-token problem in the admin, in tiny and bb .. the expandit function has no influence on it either... with ajax and without no difference.. as soon as the list of emoticons is visible somewhere the e-token error message appears... help!

**Reply by Alex-e107nl** [reply]
Oke tested on official 2.3.3 version and there's no e-token missing, just have to wait en see if it gets solved, it's above my paygrade :-)
e107.nl uses the latest git and functions also as a testsite for me so it's not a big problem, just drop the emoticons :-)

**Reply by Moc** [reply]
Can you give me step by step instructions on how to reproduce the issue on a clean install using the latest files from Github?

**Comment by Alex-e107nl** [comment]
@Moc activated emoticons, defaul pack, chatbox activated emoticons, when showing the emoticons to choose in a chatbox message the missing e-token is there, but just as i said, when i test it on a clean official 2.3.3 site the problem is not there... Tell me what you need for e107.nl and you wil get it!

**Reply by Moc** [reply]
I need more details. Explain what you mean with "missing e-token". What do you see? Are there any errors visible? What behavior do you see and what do you expect to happen instead. 

No need to dive into the code. I just need details because I don't understand what the issue is.

**Reply by Alex-e107nl** [reply]
Oke, i have activated the emoticons on e107.nl login with your account, chat with a emoticon, and inspect what happens (ff right mouseclick failures) GET 	https://e107.nl/e107_images/emotes/default/e-token - 404... Github today, PHP 8.1.26 no cahche!

Sometimes it comes directly and sometimes when you refresh the page, problem is that when you logout you get the link https://e107.nl/e107_images/emotes/default/e-token ... 404

**Reply by Jimmi08** [reply]
@Moc if it is somehow related to URLs (token or ajax), there is a problem with a backslash. e107 is adding backslash but SEF-URLS are without backslash. The issue already exists.

**Reply by Moc** [reply]
I don't have any issues using the latest Github files. 

Steps I followed:
1) Login and go to admin area
2) Activate emoticons
3) Active chatbox plugin
4) Active emoticons in chatbox plugin
5) Go to frontend website 
6) Post message in chatbox which includes an emote

All works well. 

7) Log out. All works well. No errors in console. No weird redirection or anything happening. 

What is the issue I should be seeing? What is happening for you? Does it happen also on a clean install using the bootstrap 3 theme? Or bootstrap5 theme?

**Reply by Alex-e107nl** [reply]
Please look below Tijn... and I always use the latest GitHub, and the solution is quite simple so it is true that there is a problem... Always BS5.3.2....

**Comment by Alex-e107nl** [comment]
It took 3 weeks, really, 3 WEEKS! what was the error message, missing e-token if you had the emoticons visible on your website in the chat box but also in the tinymce... so today I thought what exactly does the message give, `<a class='addEmote' data-emote =" ************************* " href="#"><img src='/e107_images/emotes/default/e-token' alt='' /></a>` but that file is not there, it's about the code.. .. **What if I add an empty file to the emoticons... and I call it 'e-token'**, let's see what happens... 

Well there is no more message or problem anymore, whether it is the right solution, I'm sure not, but I no longer get an error 404 page when I log out of e107.nl!

**Reply by Moc** [reply]
I don't understand what you are saying. Don't point towards what you think is the solution, but please describe to me what the ISSUE is you see. What behaviour happens that shouldn't happen? I don't see the code you posted anywhere. Does this issue happen on a clean installation?

**Reply by Moc** [reply]
In the source code I only see this, which is the correct display of all the emoticons in the default pack., The fact that you have a mention of 'e-token' in there makes me think that you are not using the core files but instead have made modifications yourself which mess things up. 

```html
<form id='chatbox' method='post' action='https://clean3004.test/news/?default.0.0'><div class='control-group form-group' id='chatbox-input-block'>
	<textarea placeholder="Type your message here." required class='tbox chatbox form-control input-xlarge' id='cmessage' name='cmessage' cols='20' rows='5' style='max-width:97%;  overflow: auto' onselect='storeCaret(this);' onclick='storeCaret(this);' onkeyup='storeCaret(this);'></textarea>
	<br />
	<input class='btn btn-sm btn-primary button' type='submit' id='chat_submit' name='chat_submit' value='Post new message' />
		<input class='btn btn-default btn-secondary button' type='button' style='cursor:pointer' size='30' value='Emotes' onclick="expandit('emote')" />
		<div class='well' style='display:none' id='emote'><div class='spacer'>
<a class='addEmote' data-emote=" !alien " href="#"><img src='/e107_images/emotes/default/alien.png' alt='' /></a> 
<a class='addEmote' data-emote=" !amazed " href="#"><img src='/e107_images/emotes/default/amazed.png' alt='' /></a> 
<a class='addEmote' data-emote=" !grr " href="#"><img src='/e107_images/emotes/default/angry.png' alt='' /></a> 
<a class='addEmote' data-emote=" !lol " href="#"><img src='/e107_images/emotes/default/biglaugh.png' alt='' /></a> 
<a class='addEmote' data-emote=" :D " href="#"><img src='/e107_images/emotes/default/cheesey.png' alt='' /></a> 
<a class='addEmote' data-emote=" :? " href="#"><img src='/e107_images/emotes/default/confused.png' alt='' /></a> 
<a class='addEmote' data-emote=" :(( " href="#"><img src='/e107_images/emotes/default/cry.png' alt='' /></a> 
<a class='addEmote' data-emote=" x) " href="#"><img src='/e107_images/emotes/default/dead.png' alt='' /></a> 
<a class='addEmote' data-emote=" !dodge " href="#"><img src='/e107_images/emotes/default/dodge.png' alt='' /></a> 
<a class='addEmote' data-emote=" :( " href="#"><img src='/e107_images/emotes/default/frown.png' alt='' /></a> 
<a class='addEmote' data-emote=" :@ " href="#"><img src='/e107_images/emotes/default/gah.png' alt='' /></a> 
<a class='addEmote' data-emote=" :D " href="#"><img src='/e107_images/emotes/default/grin.png' alt='' /></a> 
<a class='addEmote' data-emote=" !heart " href="#"><img src='/e107_images/emotes/default/heart.png' alt='' /></a> 
<a class='addEmote' data-emote=" :! " href="#"><img src='/e107_images/emotes/default/idea.png' alt='' /></a> 
<a class='addEmote' data-emote=" !ill " href="#"><img src='/e107_images/emotes/default/ill.png' alt='' /></a> 
<a class='addEmote' data-emote=" ~:( " href="#"><img src='/e107_images/emotes/default/mad.png' alt='' /></a> 
<a class='addEmote' data-emote=" !mistrust " href="#"><img src='/e107_images/emotes/default/mistrust.png' alt='' /></a> 
<a class='addEmote' data-emote=" :| " href="#"><img src='/e107_images/emotes/default/neutral.png' alt='' /></a> 
<a class='addEmote' data-emote=" ?! " href="#"><img src='/e107_images/emotes/default/question.png' alt='' /></a> 
<a class='addEmote' data-emote=" B) " href="#"><img src='/e107_images/emotes/default/rolleyes.png' alt='' /></a> 
<a class='addEmote' data-emote=" !sad " href="#"><img src='/e107_images/emotes/default/sad.png' alt='' /></a> 
<a class='addEmote' data-emote=" 8) " href="#"><img src='/e107_images/emotes/default/shades.png' alt='' /></a> 
<a class='addEmote' data-emote=" !shy " href="#"><img src='/e107_images/emotes/default/shy.png' alt='' /></a> 
<a class='addEmote' data-emote=" :) " href="#"><img src='/e107_images/emotes/default/smile.png' alt='' /></a> 
<a class='addEmote' data-emote=" %-6 " href="#"><img src='/e107_images/emotes/default/special.png' alt='' /></a> 
<a class='addEmote' data-emote=" :O " href="#"><img src='/e107_images/emotes/default/suprised.png' alt='' /></a> 
<a class='addEmote' data-emote=" :p " href="#"><img src='/e107_images/emotes/default/tongue.png' alt='' /></a> 
<a class='addEmote' data-emote=" ;) " href="#"><img src='/e107_images/emotes/default/wink.png' alt='' /></a> </div></div>
</div>
</form>
```

**Reply by Alex-e107nl** [reply]
Moc it is exactly as in my first post... and no it is not a clean install the website is 15 years old :-) So i have tested it on a clean install with git of 20-12-2023 and it gave me the same problem.. But as i said it is solved now. When the chatbox emoticons are looking for a file that is called e-token in the emoticons folder and it is not there.. :-)

**Reply by Alex-e107nl** [reply]
As far as i know i did not alter the chatbox or emoticons code... but it gives me a` <a class='addEmote' data-emote=" 8553049007cea8851d35eef6bd2d02df " href="#"><img src='/e107_images/emotes/default/e-token' alt='' /></a>` extra....

I will look further where it is coming from... chatbox code is old...

**Reply by Jimmi08** [reply]
@Alex-e107nl It is not looking for a file with the name e_token, e_token is the value from the form...  it is something related to safety.  You didn't fix the problem, you hid it.

**Reply by Moc** [reply]
@Jimmi08 @Alex-e107nl STOP! Please, can one of you please describe the exact steps to reproduce what you see because I really don't get it. Don't think about a solution yet. First explain the issue. 

Clean install, bootstrap 3 theme. Create a video if you want. Show me screenshots but I have no idea what is happening now. 

On a CLEAN install, with the default emoticon pack.  Check the source code. 
It should like what I posted here: https://github.com/e107inc/e107/discussions/5127#discussioncomment-7919900

Do you see something different? Another line with a new emoticon that refers to "e-token"? 

**Please don't waste time on trying to find the solution. First we need to get clarity on what the issue is.** Once I (or another developer) understands it, it is much easier to find the cause of the issue and solve it. Now it just takes longer and longer.

**Reply by Alex-e107nl** [reply]
Moc, it took me the evening but I was able to listen to the staatsbegrafenis... What did I do, installed new e107 2.3.3 original on subdomain, PHP 8.1.26 Maria database, with installation only choice for bootstrap5 and so I used that and in English. Chatbox installed, no problems, all plugins installed and tested after clearing the cache, no problems. Uninstalled Tiny4, installed Tiny 5 no problem, installed my new theme with BS 5.3.2, no problem, always cleared cache. Switched to Dutch, no problem, Uploaded the latest Git over it and did database updates, cleared the cache and still no e-token problem. Emoticon default pack was original.

Then I deleted all unnecessary plugins on e107.nl, and it turned out that there was still one for version 1... New Forum Posts ...

I have now tested on e107.nl a number of times and have not seen any more problems and I suspect it will remain that way, thanks for the help everyone!

**Reply by Alex-e107nl** [reply, marked as answer]
Today i have cleaned up the files on e107.nl, everything that's old is removed now, was a lot, i have always checked with the Fileinspector with a official release but that's not enough... This website is very very old, started with a .7 version! But now it's lean and clean :-)
Emote e-token was still there, so i have checked the database, and it was there.... (export of e107 database) 

```
INSERT INTO `e107_core` (`e107_name`, `e107_value`) VALUES
('emote_default', 'array (\n  \'alien!png\' => \'!alien\',\n  \'amazed!png\' => \'!amazed\',\n  \'angry!png\' => \'!grr !angry\',\n  \'biglaugh!png\' => \'!lol\',\n  \'cheesey!png\' => \':D :oD :-D\',\n  \'confused!png\' => \':? :o? :-?\',\n  \'cry!png\' => \'&amp;| &amp;-| &amp;o| :(( !cry\',\n  \'dead!png\' => \'x) xo) x-) x( xo( x-(\',\n  \'dodge!png\' => \'!dodge\',\n  \'frown!png\' => \':( :o( :-(\',\n  \'gah!png\' => \':@ :o@ :o@\',\n  \'grin!png\' => \':D :oD :-D\',\n  \'heart!png\' => \'!heart\',\n  \'idea!png\' => \':! :o! :-!\',\n  \'ill!png\' => \'!ill\',\n  \'mad!png\' => \'~:( ~:o( ~:-(\',\n  \'mistrust!png\' => \'!mistrust\',\n  \'neutral!png\' => \':| :o| :-|\',\n  \'question!png\' => \'?!\',\n  \'rolleyes!png\' => \'B) Bo) B-)\',\n  \'sad!png\' => \'!sad\',\n  \'shades!png\' => \'8) 8o) 8-)\',\n  \'shy!png\' => \'!shy\',\n  \'smile!png\' => \':) :o) :-)\',\n  \'special!png\' => \'%-6\',\n  \'suprised!png\' => \':O :oO :-O\',\n  \'tongue!png\' => \':p :op :-p :P :oP :-P\',\n  \'wink!png\' => \';) ;o) ;-)\',\n  \'e-token\' => \'8553049007cea8851d35eef6bd2d02df\',\n)'),
```
Don't know how it got there but i have deleted `\'e-token\' => \'8553049007cea8851d35eef6bd2d02df\',\n)` and the problem is now really solved until it gets back but now i know where to look :-)

Again thx, hope it helps others when they have the same problem..!

**Reply by Moc** [reply]
@Alex-e107nl Excellent, that's the way to do it :) Glad it got sorted

**Comment by Jimmi08** [comment]
@Moc When I played with this, I found this:
-  set chatbox type = 2 (ajax version)
- put chatbox menu on page with missing backslash f.e. /forum 
- activate emoticons, 
- go to form and inspect form URL (it is different from your actual URL)
- send the first post with emoticons - it works
 -  try to send a second post with emoticons - it doesn't work, URL was changed to /forum/  
So it breaks ajax somewhere and it can cause an issue with e_token.  (that issue can be different, it depends when ajax was broken)

I had this on e107sk.com too with emoticons, after fixing chatbox form URL (post URL), it is solved and no issues anymore.

**Reply by Moc** [reply]
That is https://github.com/e107inc/e107/issues/5143. A separate issue. Right? Please keep it out of this discussion as I don't think it is related.

**Reply by Jimmi08** [reply]
No, it is related. This is a discussion, not the issue. While discussing - trying to find a reason - I found 2 maybe related issues. So I created issues. 

I didn't see Alex issue (with e_token), neither on e107.nl...  but chatbox didn't work correctly with bootstrap5 theme on both sites. If I saw that e_token issue, I would create the issue.

**Reply by Alex-e107nl** [reply]
Guys, i wil do what Moc asked, install the git fresh from 20-12-2023, do the bootstrap 3 theme and see if its there.. in the meantime Moc, i have tried to do a pull on e107, solved the plupload problem in the admin with another language and made the emoticons button the same as the one next to it, can you look into it? ( [Link](https://github.com/e107inc/e107/commit/105cf1aea8d9097215639ce151665beca083cbbf) ) ( [Link](https://github.com/e107inc/e107/commit/3d68525abeb2ad5bd0861ca78cd23cd8ad47ad8c) )
